packages/lsp/src/adapters/adapter.ts

```diff
@@ -592,6 +599,32 @@ export abstract class WidgetLSPAdapter<T extends IDocumentWidget>
   private _isConnected: boolean;
   private _updateFinished: Promise<void>;
   private _virtualDocument: VirtualDocument | null = null;
+  private _editorToAdapter: WeakMap<Document.IEditor, EditorAdapter>;
+
+  private _onEditorAdded(
+    sender: WidgetLSPAdapter<T>,
+    change: IEditorChangedData
+  ): void {
+    const { editor } = change;
+    const editorAdapter = new EditorAdapter({
+      editorReady: editor.ready,
+      getEditor: editor.getEditor,
+      path: this.documentPath,
+      adapter: this,
```

Comment:
Since `EditorAdapter` has also `Adapter` in its name
```suggestion
      widgetAdapter: this,
```